src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java

```diff
@@ -292,6 +296,35 @@ public void onBlockBreak(BlockBreakEvent e) {
         }
     }
 
+    @EventHandler
+    public void onExperienceReceive(PlayerExpChangeEvent e) {
+        if (e.getAmount() <= 0 || !Talisman.checkFor(e, SlimefunItems.TALISMAN_WISE)) {
+            return;
+        }
+
+        e.setAmount(e.getAmount() * 2);
+    }
+
+    @EventHandler
+    public void onPlayerDeath(PlayerDeathEvent e) {
+        Player player = e.getEntity().getPlayer();
+        DamageCause dmgCause = player.getLastDamageCause().getCause();
+
+        if (dmgCause != DamageCause.VOID || !Talisman.checkFor(e, SlimefunItems.TALISMAN_RESURRECTED)) {
+            return;
+        }
+
+        Location loc = e.getEntity().getPlayer().getBedSpawnLocation();
+        if (loc == null)
+            loc = player.getLocation().getWorld().getSpawnLocation();
+
+        final Location respawnLocation = loc;
+
+        Bukkit.getScheduler().runTaskLater(SlimefunPlugin.instance(), () -> {
+            player.teleport(respawnLocation);
+        }, 2);
```

Comment:
Use `SlimefunPlugin.runSync`

```suggestion
        SlimefunPlugin.runSync(() -> {
            player.teleport(respawnLocation);
        }, 2);
```